Sentinel/Classes/Core/Internal/SentinelInternal.swift

```diff
@@ -26,37 +28,75 @@ extension Sentinel {
 }
 
 private extension Sentinel {
-    func createTabBarControllers(with tools: [Tool], viewController: UIViewController) -> [UIViewController] {
-        var tabBarControllers = tools.filter { tool in
-            return tool.type == .application ||
-            tool.type == .permissions ||
-            tool.type == .performance
-        }.map { $0.createViewController(on: viewController) }
-
-        let toolItems = tools.filter { tool in
-            return tool.type != .application &&
-            tool.type != .permissions &&
-            tool.type != .performance
-        }
-
-        let toolsViewController = createToolsController(with: toolItems, viewController: viewController)
-        let deviceViewController = createDeviceViewController()
-
-        tabBarControllers.insert(deviceViewController, at: 0)
-        tabBarControllers.insert(toolsViewController, at: 2)
+    func createTabBarControllers(with tools: [Tool], optionSwitchItems: [OptionSwitchItem], viewController: UIViewController) -> [UIViewController] {
+        var tabBarControllers: [UIViewController] = []
+        tabBarControllers.append(createDeviceViewController())
+        tabBarControllers.append(createApplicationViewController())
+        tabBarControllers.append(createToolsController(with: tools, viewController: viewController))
+        tabBarControllers.append(createPreferencesViewController(optionSwitchItems))
+        tabBarControllers.append(createPerformanceViewController())
         return tabBarControllers
     }
 
     func createToolsController(with tools: [Tool], viewController: UIViewController) -> UIViewController {
-        let navigationItems = tools.map { NavigationToolTableItem(title: $0.name, navigate: $0.presentPreview(from:)) }
+        let navigationItems = tools
+            .map { NavigationToolTableItem(title: $0.name, navigate: $0.presentPreview(from:)) }
         let section = ToolTableSection(title: "Tools", items: navigationItems)
-        let toolsViewController = ToolTable(name: "Tools", sections: [section]).createViewController(on: viewController)
-        toolsViewController.tabBarItem = UITabBarItem(title: "Tools", image: UIImage.Sentinel.tools.resize(), tag: 2)
+        let toolTable = ToolTable(name: "Tools", sections: [section])
+        let toolsViewController = SentinelTableViewController.create(with: toolTable)
+        toolsViewController.tabBarItem = UITabBarItem(
+            title: "Tools",
+            image: UIImage.resize(UIImage.loadImageFromBundle(name: "tools")),
+            tag: 2
+        )
         return toolsViewController
     }
 
     func createDeviceViewController() -> UIViewController {
-        let deviceInfoItem = DeviceTool()
-        return deviceInfoItem.createViewController()
+        let deviceInfoItem = DeviceTool().tool
```

Comment:
This feels a bit weird. Can it be:
```suggestion
        let deviceInfoItem = DeviceTool()
```
I would expect usage like this 😄